### PR TITLE
feat: Add documentation URL alongwith the error response to offer more info about the exception

### LINKED ISSUE #10004: Verify git documentation links are not hardcoded on the frontend
- [ ] ssh url input
- [ ] ssh key box
- [ ] conflicts found: (deploy / merge tabs and bottom bar popup)
- [ ] disconnect popup
- [ ] repo limit reached modal
- [ ] error while connecting at the connection tab
- [ ] push error: pending upstream changes at the deploy tab

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> When server throws the error, in some cases like git exception it is possible to offer a step-by-step guide on how to resolve the error. This PR introduces the `referenceDoc` field in the error response which client can consume to offer read more section for errors. 

Fixes #10004

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> JUnit TC
> Manual test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-git/src/main/java/com/appsmith/git/service/GitExecutorImpl.java [hunk 1]
```diff
@@ -61,9 +62,7 @@
 
     public static final DateTimeFormatter ISO_FORMATTER = DateTimeFormatter.ISO_INSTANT.withZone(ZoneId.from(ZoneOffset.UTC));
 
-    private final Scheduler scheduler = Schedulers.elastic();
```
Comment (abhvsn) [inline app/server/appsmith-git/src/main/java/com/appsmith/git/service/GitExecutorImpl.java:64]: This has been deprecated so modified to `boundedElastic()`

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ErrorReferenceDocUrl.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import lombok.Getter;
+
+@Getter
+public class ErrorReferenceDocUrl {
+    public static final String GIT_MERGE_CONFLICT = "https://www.atlassian.com/git/tutorials/making-a-pull-request";
+    public static final String GIT_PULL_CONFLICT = "https://www.atlassian.com/git/tutorials/making-a-pull-request";
+    public static final String GIT_DEPLOY_KEY = "https://docs.github.com/en/developers/overview/managing-deploy-keys#deploy-keys";
+    public static final String FILE_PATH_NOT_SET = "https://docs.appsmith.com/setup/instance-configuration#configuring-docker-installations";
+    public static final String GIT_UPSTREAM_CHANGES = "https://git-scm.com/docs/git-pull";
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ErrorReferenceDocUrl.java:11]: These are temporary docs we are using as a ref till the time we have our own docs for these specific cases

File: app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/GitControllerCE.java [hunk 3]
```diff
@@ -163,7 +163,7 @@ public GitControllerCE(GitService service) {
     }
 
     @PostMapping("/merge/{defaultApplicationId}")
-    public Mono<ResponseDTO<GitPullDTO>> merge(@PathVariable String defaultApplicationId,
+    public Mono<ResponseDTO<MergeStatusDTO>> merge(@PathVariable String defaultApplicationId,
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/GitControllerCE.java:166]: As we were not using the application within `GitPullDTO` it made sense to update the response to `mergeStatusDTO` here

**Review by AnaghHegde — APPROVED** [review]
LGTM

**Pull request merged**